dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -48,6 +50,18 @@
     private final URLMapAPIImpl urlMapAPIImpl;
     private final LanguageWebAPI languageWebAPI;
 
+    @FunctionalInterface
+    private interface SearchPageFunction {
+        Optional<HTMLPageUrl> search(final PageContext context,
+                             final Host host,
+                             final HttpServletRequest request) throws DotDataException, DotSecurityException;
+    }
+
+    final List<SearchPageFunction> pageSearchers = list(
```

Comment:
if you are not planning to expose this, you can set to private and static to get init just once

Reply:
just private for now

https://github.com/dotCMS/core/pull/17859/commits/d8d0ac8874bfdf61503d97d02689d958d2fa12d4#diff-051d49b06feb0c7462b349b6b61988a8R58